agents/grpc/src/grpc_agent.cc

```diff
@@ -251,6 +257,19 @@ void PopulateInfoEvent(grpcagent::InfoEvent* info_event,
         (*body->mutable_versions())[key] = value.get<std::string>();
       }
     }
+    
+    // Get Kernel version
+    std::string kernel_version = "unknown";
+    #ifdef __linux__
+    // The presence of /sys/kernel/btf indicates kernel support for BTF (BPF Type Format),
+    // often necessary for modern eBPF features. See https://www.kernel.org/doc/html/latest/bpf/btf.html
+    struct utsname kernel_info;
+    if (uname(&kernel_info) == 0) {
```

Comment:
We should check if this would work on all platforms. I think there might be a safer approach, idk.

Reply:
We should probably start by using the same implementation as uv has. It covers more cases. Also, I think storing the version in an unsigned makes it easier to work with:

https://github.com/libuv/libuv/blob/086da2c458d3f58e1a60a4b314b4bbaa0fee4d6d/src/unix/linux.c#L302